plugins/flatpak/gs-flatpak.c

```diff
@@ -1665,9 +1665,14 @@ gs_flatpak_add_updates_pending (GsFlatpak *self, GsAppList *list,
 
 		update_priority = get_app_update_priority (app);
 
-		main_app = get_real_app_for_update (self, app, cancellable, error);
-		if (main_app == NULL)
-			return FALSE;
+		main_app = get_real_app_for_update (self, app, cancellable, &error_local);
+		/* do not fail if one of the updates cannot be done (otherwise the update
+		 * list will be empty), just inform the user */
+		if (main_app == NULL) {
+			g_warning ("Failed to get real app when updating %s: %s",
+				   gs_app_get_unique_id (app), error_local->message);
+			continue;
```

Comment:
What if all the `get_real_app_for_update()` calls fail (for all the `xrefs`), and `list` ends up empty that way? Should `gs_flatpak_add_updates_pending()` return an error in that case?

Reply:
I pondered exactly that, since I think we take a similar approach elsewhere in g-s IIRC. However, due to the limited time for testing, the fact that we do get the messages in the log (in case we need a detailed report from users), and that this code will be modified by the rebase, I think we should proceed without that.
Does this make sense?

Reply:
> However, due to the limited time for testing, the fact that we do get the messages in the log (in case we need a detailed report from users), and that this code will be modified by the rebase, I think we should proceed without that.
> Does this make sense?

IMO it’s less a case of whether we get messages in the log (though that is good), and more a case of whether the calling code expects an empty list to ever be returned if the function doesn’t set an error.

Reply:
That is okay AFAICS because it will just assume there was no updates at all, and the same could happen before when there were no updates to be done (the list is passed as parameter BTW, and it's okay to not touch it).

Reply:
OK.